craft_parts/packages/__init__.py

```diff
@@ -49,4 +52,20 @@ def _get_repository_for_platform() -> type["BaseRepository"]:
     return DummyRepository
 
 
-Repository = _get_repository_for_platform()
+class _RepositoryProxy:
```

Comment:
I'm ok with this implementation but I'm going to suggest an alternative for you to consider: Instead of proxys and deferred calls, how about adding some top-level functions `packages.refresh_packages_list()`, `packages.download_packages()` etc that get the correct repo class then call the method on it?

Reply:
That sounds like a worthy refactoring. I think we should go for that, but in a separate PR -- doing that now could disrupt a lot of code especially in tests. Created issue https://github.com/canonical/craft-parts/issues/1259.